### ISSUE TITLE
Build Model structure

### ISSUE THREAD

**Issue by Cogdof** [Cogdof/2019_Capstone-design#4]
Build Model structure

- CNN
- L:oss function : Softmax
- Input : 1 PNG image,
- Output : Multiple recommended images. 2 or 3

**Comment by Cogdof** [comment]
### Model  '1029_79%.h5' result
- Train 336000 images
- Valid 28000
- Test 5600
- Classification 56 labels, 

![image](https://user-images.githubusercontent.com/36148762/67753425-06e1ed80-fa78-11e9-9ce4-f7c3b43314f3.png)


### Result : Need to Boost model's classification performance.
Case 1,  Add models's layer. 
Case 2, Get additional data set to train.

@KangSinWon

**Comment by Cogdof** [comment]
### Model '1030_5layer_200step_100epoch.h5' result

- Train 336000 images
- Valid 28000
- Test 5600
- Classification 56 labels,
- **Fix layer 3 to 5**

![image](https://user-images.githubusercontent.com/36148762/67855746-5f87b800-fb56-11e9-9418-c27f87485ac7.png)
![1030_5layer_200step_100epoch](https://user-images.githubusercontent.com/36148762/67855705-4a128e00-fb56-11e9-9271-1b247032d0d4.png)
![1030_5layer_200step_100epoch_acc](https://user-images.githubusercontent.com/36148762/67855706-4a128e00-fb56-11e9-8607-2dc207291fd1.png)

Result : 
Case 1, Add models's layer.  <- Train faster then before.

Next to do : **Case 2, Get additional data set to train.**

@KangSinWon

**Comment by Cogdof** [comment]
### Performance boost
Reduce feature vector
1. make channel RGB ->1 channel.
2. resize 255,255 -> 48,48 or more smaller size.

Refer to the model written in the paper
http://cs229.stanford.edu/proj2018/report/98.pdf
https://uu.diva-portal.org/smash/get/diva2:1218490/FULLTEXT01.pdf
http://openaccess.thecvf.com/content_CVPR_2019/papers/Dey_Doodle_to_Search_Practical_Zero-Shot_Sketch-Based_Image_Retrieval_CVPR_2019_paper.pdf

**Comment by Cogdof** [comment]
Model ' v2.00.h5 ' result

Classification 8 labels,


![image](https://user-images.githubusercontent.com/36148762/68122463-b245e300-ff4d-11e9-8f03-819ebae5a672.png)
![v2 00_acc](https://user-images.githubusercontent.com/36148762/68122502-c8ec3a00-ff4d-11e9-9a62-f96d0378c3f1.png)
![v2 00_loss](https://user-images.githubusercontent.com/36148762/68122503-c8ec3a00-ff4d-11e9-9f4c-b9041f594d43.png)

**Comment by Cogdof** [comment]
# Model  result

**Train 1300000 images : each label has 13000 images
Valid 1000000
Test 1000000
Classification 100 labels, <- separate confuse labels** 

## ' v2.02.h5 '
![v2 02_summary](https://user-images.githubusercontent.com/36148762/68363501-60c86e80-016e-11ea-92e3-2634eae807c1.png)
![v2 02_acc](https://user-images.githubusercontent.com/36148762/68363474-49898100-016e-11ea-9969-775449761a34.png)
![v2 02_loss](https://user-images.githubusercontent.com/36148762/68363475-4a221780-016e-11ea-8db7-d83134aa63a8.png)


<hr/>

## ' v2.15.h5 '
![v2 15_summary](https://user-images.githubusercontent.com/36148762/68363498-5dcd7e00-016e-11ea-851d-252b7cf71a58.png)

![v2 15_acc](https://user-images.githubusercontent.com/36148762/68363491-58703380-016e-11ea-9814-4f0549ce755c.png)
![v2 15_loss](https://user-images.githubusercontent.com/36148762/68363492-58703380-016e-11ea-8c06-d3d24e205919.png)
